packages/flutter_availability/lib/src/ui/widgets/template_week_overview.dart

```diff
@@ -0,0 +1,178 @@
+import "package:flutter_availability/src/ui/view_models/template_daydata_view_model.dart";
+import "package:flutter_availability/src/ui/view_models/week_template_view_models.dart";
+import "package:flutter_availability/src/ui/widgets/calendar_grid.dart";
+import "package:flutter_availability/src/util/scope.dart";
+import "package:flutter_availability_data_interface/flutter_availability_data_interface.dart";
+
+/// Shows an overview of the template of a week before saving it
+class TemplateWeekOverview extends StatelessWidget {
+  ///
+  const TemplateWeekOverview({
+    required this.template,
+    super.key,
+  });
+
+  /// The template to show
+  final WeekTemplateViewModel template;
+
+  @override
+  Widget build(BuildContext context) {
+    var theme = Theme.of(context);
+    var textTheme = theme.textTheme;
+    var availabilityScope = AvailabilityScope.of(context);
+    var options = availabilityScope.options;
+    var translations = options.translations;
+    var colors = options.colors;
+
+    var dayNames = getDaysOfTheWeekAsStrings(translations, context);
+
+    var templateData = template.data;
+
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Text(
+          translations.weekTemplateOverviewTitle,
+          style: textTheme.titleMedium,
+        ),
+        const SizedBox(height: 8),
+        DecoratedBox(
+          decoration: BoxDecoration(
+            color: colors.templateWeekOverviewBackgroundColor ??
+                theme.colorScheme.primaryContainer,
+            borderRadius: BorderRadius.circular(10),
+            border: Border.all(
+              color: theme.dividerColor,
+            ),
+          ),
+          child: Column(
+            children: [
+              for (var day in WeekDay.values) ...[
+                _TemplateDayDetailRow(
+                  dayName: dayNames[day.index],
+                  dayData:
+                      templateData.containsKey(day) ? templateData[day] : null,
+                  isOdd: day.index.isOdd,
+                ),
+              ],
+            ],
+          ),
+        ),
+      ],
+    );
+  }
+}
+
+class _TemplateDayDetailRow extends StatelessWidget {
+  const _TemplateDayDetailRow({
+    required this.dayName,
+    required this.dayData,
+    required this.isOdd,
+  });
+
+  /// The name of the day
+  final String dayName;
+
+  /// There odd rows do not have a background color
+  /// This causes a layered effect
+  final bool isOdd;
+
+  /// The data of the day
+  final DayTemplateDataViewModel? dayData;
+
+  @override
+  Widget build(BuildContext context) {
+    var theme = Theme.of(context);
+    var textTheme = theme.textTheme;
+    var availabilityScope = AvailabilityScope.of(context);
+    var options = availabilityScope.options;
+    var translations = options.translations;
+
+    var startTime = dayData?.startTime;
+    var endTime = dayData?.endTime;
+    var dayPeriod = startTime != null && endTime != null
```

Comment:
This seems to have a semantic value of a completed dayData, maybe that should be clear in this check.

Also, maybe you should not use a ternary here.